src/main/kotlin/net/weavemc/loader/api/event/EventTypes.kt

```diff
@@ -54,10 +98,34 @@ public class MouseEvent : CancellableEvent() {
 
 }
 
+/**
+ * This cancellable event is called when your client receives a chat message from the server.
+ *
+ * If cancelled, the message will not be displayed in chat.
+ * @param message The message being received, in the form of a [Chat Component][IChatComponent].
+ */
 public class ChatReceivedEvent(public val message: IChatComponent) : CancellableEvent()
+
+/**
+ * This cancellable event is called when your client sends a chat message to the server.
+ *
+ * If cancelled, the message will not be sent, but still added to your chat history.
```

Comment:
It's not added to the chat history, it will cancel all processing done with the chat message.